apps/bfd-data-npi/src/main/java/gov/cms/bfd/data/npi/utility/DataUtilityCommons.java

```diff
@@ -271,45 +315,42 @@ private static void buildNPIResource(
    */
   private static void convertNpiDataFile(Path convertedNpiDataFile, Path originalNpiDataFile)
       throws IOException {
+
+    Map<String, String> taxonomyMap = DataUtilityCommons.processTaxonomyDescriptions();
     // convert file format from cp1252 to utf8
     CharsetDecoder inDec =
         Charset.forName("windows-1252")
             .newDecoder()
             .onMalformedInput(CodingErrorAction.REPORT)
             .onUnmappableCharacter(CodingErrorAction.REPORT);
-
     CharsetEncoder outEnc =
         StandardCharsets.UTF_8
             .newEncoder()
             .onMalformedInput(CodingErrorAction.REPORT)
             .onUnmappableCharacter(CodingErrorAction.REPORT);
-
     try (FileInputStream is = new FileInputStream(originalNpiDataFile.toFile().getAbsolutePath());
         BufferedReader reader = new BufferedReader(new InputStreamReader(is, inDec));
         FileOutputStream fw =
             new FileOutputStream(convertedNpiDataFile.toFile().getAbsolutePath());
         BufferedWriter out = new BufferedWriter(new OutputStreamWriter(fw, outEnc))) {
-
-      // Get indexes according to header
-      String line = reader.readLine();
-      String[] fields = line.split(",");
-      Map<String, Integer> indexes = getIndexNumbers(fields);
-      Integer npiIndex = getIndexNumberForField(indexes, "NPI");
-      Integer entityTypeCodeIndex = getIndexNumberForField(indexes, "Entity Type Code");
-      Integer orgNameIndex =
-          getIndexNumberForField(indexes, "Provider Organization Name (Legal Business Name)");
-
-      while ((line = reader.readLine()) != null) {
-        fields = line.split(",");
-
-        String orgName = fields[orgNameIndex].trim().replace("\"", "");
-        String npi = fields[npiIndex].trim().replace("\"", "");
-        String entityTypeCode = fields[entityTypeCodeIndex].trim().replace("\"", "");
-
+      CSVParser csvParser =
+          new CSVParser(
+              reader,
+              CSVFormat.DEFAULT.withFirstRecordAsHeader().withIgnoreHeaderCase().withTrim());
+      for (CSVRecord csvRecord : csvParser) {
+        String orgName = csvRecord.get(PROVIDER_ORGANIZATION_NAME_FIELD);
+        String npi = csvRecord.get(NPI_FIELD);
+        String entityTypeCode = csvRecord.get(ENTITY_TYPE_CODE_FIELD);
+        String taxonomyCode = csvRecord.get(TAXONOMY_CODE_FIELD);
         // entity type code 2 is organization
-        if (!Strings.isNullOrEmpty(entityTypeCode) && Integer.parseInt(entityTypeCode) == 2) {
-          out.write(npi + "\t" + orgName);
-          out.newLine();
+        if (!Strings.isNullOrEmpty(entityTypeCode)) {
+          if (Integer.parseInt(entityTypeCode) == 2 && !Strings.isNullOrEmpty(orgName)) {
+            out.write(npi + "\t" + orgName);
+            out.newLine();
+          } else if (!Strings.isNullOrEmpty(taxonomyCode)) {
```

Comment:
This should resolve the empty rows in the constructed file.

Reply:
What's the reason that some of the org names are empty?

Reply:
Makes sense, thanks!

Reply:
Actually, I don't think this is always the case. I added this line
```java
 if (Integer.parseInt(entityTypeCode) == 2 && Strings.isNullOrEmpty(orgName)) {
            System.out.println("empty org");
 }
```
and it got a few hits before I stopped it.